### ISSUE TITLE
inconsistent of single width rounding mode description

### ISSUE THREAD

**Issue by HanKuanChen** [riscv/riscv-v-spec#299]
According to [rounding mode](https://github.com/riscv/riscv-v-spec/blob/master/v-spec.adoc#35-vector-fixed-point-rounding-mode-register-vxrm), rounded result is `(v >> d) + r`.

However, I notice most of single width instructions is `(v + r) >> d`.
E.g., [averaging add](https://github.com/riscv/riscv-v-spec/blob/master/v-spec.adoc#132-vector-single-width-averaging-add-and-subtract).
```
# result = (src1 + src2 + round) >> 1;
```

I think it should be
```
# result = ((src1 + src2) >> 1) + round;
```
?

**Comment by JamesKenneyImperas** [comment]
I agree. Also note that the expression:

`    ((src1 + src2) >> 1)`

must be evaluated with operands sign-extended to SEW+1 bits for the result to be correct, and this is not made clear. For example, with SEW=8, an averaging add of -128 and -128 (i.e. 0x80 and 0x80) should produce a result 0x80, not 0x00 (which would be the case if not extended).

See also issue #277.

**Comment by kasanovic** [comment]
Rounding description issues fixed in commit db4c604.

**Comment by kasanovic** [comment]
#256 discusses unsigned average add/sub instructions so closing this issue here.
